### PR TITLE
CB-8420 'cordova plugin add' should retrieve version, url or local folder from config.xml if none is specified

### PR DESCRIPTION
**Pull request by omefire**
CB-8420 'cordova plugin add' should retrieve version, url or local folder from config.xml if none is specified

### CODE REVIEW

**Comment by omefire** [comment]
@vladimir-kotikov, @agrieve and @gorkem , please review.

**Comment by gorkem** [comment]
-1 since we agreed to have auto restore for plugins I do not think this feature will be used at all.

**Comment by omefire** [comment]
Well, if you review the document we all agreed on, it mentions this. Plus, this feature is already in for platforms as well. I'm trying to mirror it : https://docs.google.com/document/d/1qKjhzSf48ybGg2GFZPtjXP8dkF4Z5Jnc7FU41V3-jFc/edit#heading=h.tsqn5em3043l

**Comment by omefire** [comment]
@gorkem Also, this feature does not prevent the other one you mentionned.

**Comment by vladimir-kotikov** [comment]
I'm not aware of overall design of save/restore features, but code LGTM.

File: cordova-lib/src/cordova/plugin.js [hunk 1]
```diff
@@ -114,6 +116,22 @@ module.exports = function plugin(command, targets, opts) {
...
+                        var parts = target.split('@');
+                        var id = parts[0];
+                        var version = parts[1];
+
+                        // If no version is specified, retrieve the version from config.xml
+                        if(!version && !cordova_util.isUrl(id) && !cordova_util.isDirectory(id)){
+                            events.emit('verbose', 'no version specified, retrieving version from config.xml');
+                            var ver = getVersionFromConfigFile(id, cfg);
+
+                            if( cordova_util.isUrl(ver) || cordova_util.isDirectory(ver) ){
```
Comment (agrieve) [inline cordova-lib/src/cordova/plugin.js]: isDirectory don't check for null/undefined. Should either add that to isDirectory, or add check here.

**Comment by agrieve** [comment]
lgtm save one nit. I think this behaviour is fine. Normal case will be to use add --save anyways.

File: cordova-lib/src/cordova/plugin.js (hunk 1 above)
Comment (omefire) [inline cordova-lib/src/cordova/plugin.js]: @agrieve, isDirectory does handle the null/undefined cases. I just tested it. Am I missing something ?

**Pull request closed without merging**